transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -1047,6 +1052,12 @@ protected Executor prepareToClose() {
         }
     }
 
+    private static final ClosedChannelException CLOSE_CHANNEL_EX = new ClosedChannelException() {
```

Comment:
we can not make this a static instance as the user could access the exception and call `addSuppressed(...)` on it. This object would then never been garbage collected and so may cause OOME.

Reply:
That's a problem. Or we cloud disabled addSuppressed if security allow.
Or default is new instance and add system property can trun the control flow exception as singleton:)